infra/terraform/modules/aws_account_logging/cloudtrail/cloudtrail_to_es.py

```diff
@@ -47,28 +49,41 @@ def post_to_elasticsearch(doc):
 
     # TODO handle errors
     response = requests.post(
-        elasticsearch_url(os.environ),
+        elasticsearch_url(os.environ, get_pipeline(doc)),
         auth=elasticsearch_auth_header(os.environ),
         json=doc)
 
     LOG.debug('Elasticsearch response: {status} - {text}'.format(
         status=response.status_code,
         text=response.text))
 
+def get_pipeline(doc):
+    # sourceIPAddress may not be present, or may contain a hostname,
+    # which causes an ElasticSearch error - so only specify the geoip
+    # pipeline if an actual IP address is present
+    try:
+        ipaddress.ip_address(doc['sourceIPAddress'])
+        return 'cloudtrail-geoip'
+    except (KeyError, ValueError):
+        return None
```

Comment:
I'd be tempted to separate the test for a valid IP address from setting the URL parameter, for a more self documenting approach:
```python
def is_ip_address(source_ip_address):
    try:
        ipaddress.ip_address(source_ip_address)

    except ValueError:
        return False

    return True
```
and then
```python
params = {}

if is_ip_address(doc.get('sourceIPAddress')):
    params['pipeline'] = 'cloudtrail-geoip'

requests.post(elasticsearch_url(os.environ), params=params, ...)
```